webapp/src/js/services/contact-view-model-generator.js

```diff
@@ -152,15 +152,27 @@ angular.module('inboxServices').factory('ContactViewModelGenerator',
     };
 
     const getChildren = contactId => DB()
-      .query('medic-client/contacts_by_parent', { key: contactId, include_docs: true })
+      .query('medic-client/contacts_by_parent', { startkey: [contactId], endkey: [contactId, {}], include_docs: true })
       .then(response => response.rows);
 
-    var loadChildren = function(model) {
+    const getChildrenOfTypePerson = contactId => DB()
+      .query('medic-client/contacts_by_parent', { key: [contactId, 'person'], include_docs: true })
+      .then(response => response.rows);
```

Comment:
There's a bit of unnecessary duplication here. How about rewriting it like this...

```js
const getChildren = (contactId, { getChildPlaces }) => {
  const options = { include_docs: true };
  if (getChildPlaces) {
    // get all types
    options.startkey = [ contactId ];
    options.endkey = [ contactId, {} ];
  } else {
    // just get people
    options.key = [ contactId, 'person' ];
  }
  return DB().query('medic-client/contacts_by_parent', options)
    .then(response => response.rows);
};
```

Then the code below becomes much simpler:

```js
const loadChildren = (model, options={}) => {
  ...
  return getChildren(model.doc._id, options)
    .then(splitContactsByType)
    ...
}
```